src/rust/logger/src/backend/file/sender.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright 2021 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+use crate::*;
+
+/// A `FileLogSender` implements `Log` and can be registered as the main logger.
```

Comment:
To me the destination of the log messages (a file, a Kafka producer, another socket...) should probably only matter to the receiver, which uses that info to carry out the `flush` operation. While there may be some implications to the sender (e.g. we may want to serialize things differently based on destination), characterizing senders directly on that seems like leaky abstraction.

Reply:
Maybe we can have `PlainTextSender` `BinarySender` etc

Reply:
This is fair. Good suggestion for cleaning that up. Can probably make the Sampling* variant generic across some type implementing Log.